tests/v2/validation/provisioning/rke1/support_matrix_test.go

```diff
@@ -0,0 +1,179 @@
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SupportMatrixTestSuite struct {
+	suite.Suite
+	session            *session.Session
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	ns                 string
+	kubernetesVersions []string
+	cnis               []string
+	nodeProviders      []string
+	advancedOptions    provisioning.AdvancedOptions
+}
+
+func (s *SupportMatrixTestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *SupportMatrixTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	s.kubernetesVersions = clustersConfig.RKE1KubernetesVersions
+	s.cnis = clustersConfig.CNIs
+	s.nodeProviders = clustersConfig.NodeProviders
+	s.advancedOptions = clustersConfig.AdvancedOptions
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+
+	s.kubernetesVersions, err = kubernetesversions.Default(s.client, clusters.RKE1ClusterType.String(), s.kubernetesVersions)
+	require.NoError(s.T(), err)
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(s.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(s.T(), err)
+
+	s.standardUserClient = standardUserClient
+}
+
+func (s *SupportMatrixTestSuite) TestSupportMatrixChecks() {
+	require.GreaterOrEqual(s.T(), len(s.cnis), 1)
+
+	nodeRoles := []string{
+		"--etcd --controlplane --worker",
+	}
+
+	tests := []struct {
+		name      string
+		nodeRoles []string
+		client    *rancher.Client
+	}{
+		{"1 Node all roles " + provisioning.StandardClientName.String(), nodeRoles, s.standardUserClient},
+	}
+	var name string
+	for _, tt := range tests {
+		testSession := session.NewSession()
+		defer testSession.Cleanup()
+
+		client, err := tt.client.WithSession(testSession)
+		require.NoError(s.T(), err)
+
+		for _, nodeProviderName := range s.nodeProviders {
+			externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviderName)
+			providerName := " Node Provider: " + nodeProviderName
+			for _, kubeVersion := range s.kubernetesVersions {
+				name = tt.name + providerName + " Kubernetes version: " + kubeVersion
+				for _, cni := range s.cnis {
+					name += " cni: " + cni
```

Comment:
nit; there's something strange with the naming going on, do you know if you can fix it? 
```
...-rancher2-1_cni:_flannel_cni:_canal
```

Reply:
Noticed that as well. Without just removing the cni portion of the name, not sure. Our other tests follow the same format so I think they would see the same thing as well?

Reply:
oh I think I see why, can you change it from `+=` to `name = tt.name + providerName + " Kubernetes version: " + kubeVersion + " cni: " + cni` and remove the previous loop's name definition?

Reply:
Nice, sure I'll try that locally. If that works, I'll push a commit and get a new freeform job so we can be safe.

Reply:
That worked, I'll push a new commit.